src/main/java/com/kodilla/ecommercee/service/CartDbService.java

```diff
@@ -0,0 +1,110 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.dao.CartDao;
+import com.kodilla.ecommercee.dao.OrderDao;
+import com.kodilla.ecommercee.dao.ProductDao;
+import com.kodilla.ecommercee.dao.UserDao;
+import com.kodilla.ecommercee.domain.Cart;
+import com.kodilla.ecommercee.domain.Order;
+import com.kodilla.ecommercee.domain.Product;
+import com.kodilla.ecommercee.domain.User;
+import com.kodilla.ecommercee.exception.CartNotFoundException;
+import com.kodilla.ecommercee.exception.ProductNotFoundException;
+import com.kodilla.ecommercee.exception.UserNotAuthenticatedException;
+import com.kodilla.ecommercee.exception.UserNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class CartDbService {
+
+    private final CartDao cartDao;
+    private final ProductDao productDao;
+    private final UserDao userDao;
+    private final OrderDao orderDao;
+    private final UserAuthenticator authenticator;
+
+    public void createCart(Long userId, String userKeyToCheck) throws UserNotFoundException,
```

Comment:
w tych miejscach chyba by styknęło jakby było throws Exception, bo oba które mogą być rzucone po nim dziedziczą

Reply:
Z tego, co wiem to raczej nie jest dobra praktyka, bo nie pozwala Ci na odpowiednie obsłużenie złapanych wyjątków, bo w ten sposób obcinasz sobie informacje. W naszym przypadku głównie rzucamy te wyjątki i nic z nimi nie robimy ale w rzeczywistości musielibyśmy to bardziej ogarnąć. 
Przykład pierwszy z brzegu po wpisaniu w google:
https://stackify.com/common-mistakes-handling-java-exception/

Reply:
ma to sens :)